Source/Widgets/InfoBox/InfoBox.js

```diff
@@ -141,6 +146,36 @@ click: function () { closeClicked.raiseEvent(this); }');
 
     defineProperties(InfoBox.prototype, {
         /**
+         * A flag that allows scripts to run inside the infobox from the string.
+         * Due to security issues default is false. It must be explicitly set during or after infobox creation.
+         * @memberof InfoBox.prototype
+         *
+         * @type {Boolean}
+         *
+         * @example
+         * var viewer = new Cesium.Viewer('cesiumContainer');
+         * viewer.infoBox.allowScripts = true;
+         *
+         * * @example
+         * var infoBox = new Cesium.InfoBox('infoBoxContainer', true);
+         */
+        allowScripts: {
```

Comment:
We expose the `frame` property so that people can add or remove whatever sandbox attributes they want.  Adding a new property here means `this._allowScripts` may not actually match the underlying sandbox attribute.  I would suggest we remove this property and just have users continue to manipulate `frame.sandbox` directly.  This will also prevent this property from becoming out out of date if the underlying sandbox changes.

Likewise, elsewhere in this file, where we check the value of `allowScripts`, we should instead check the current attributes directly.

Reply:
`allowScripts` is much stricter than just the sandbox attributes it changes.  It also uses the contextualFragment option, which actually makes scripts work.  Just another layer of "protection".  Do we really want to lost it?

Reply:
It's not actually a layer of protection.  If `allow-scripts` is enabled, then this check won't prevent someone from injecting malicious code into the application whether you're using `contextualFragment` or not.  Using `contextualFragment` just makes it (hopefully) work in all of the expected cases.

For example, here's code that runs javascript inside of a description in master without this pull request:

```js
var viewer = new Cesium.Viewer('cesiumContainer');
viewer.infoBox.frame.sandbox.add('allow-scripts');
viewer.infoBox.frame.sandbox.add('allow-modals');

viewer.entities.add({
    position: Cesium.Cartesian3.fromDegrees(0,0,0),
    description: "<img src='x' onerror='alert(1)'>",
    point:{
        pixelSize : 50
    }
});
```

Reply:
Yea - I know about the `image` trick.  I meant no `<script>` runs :)
In any event, if a developer doesn't sanitize user input that is injected to the DOM, the app will be vulnerable.